frontend/javascripts/oxalis/view/jobs/train_ai_model.tsx

```diff
@@ -67,106 +201,122 @@ export function TrainAiModelTab({ onClose }: { onClose: () => void }) {
     onClose();
   };
 
-  const colorLayers = getColorLayers(dataset);
-  const colorLayer = colorLayers[0];
+  if (annotationsWithDatasets.length === 0 && onAddAnnotationsWithDatasets != null) {
+    return (
+      <Form form={form} layout="vertical">
+        <AnnotationsCsvInput onAdd={onAddAnnotationsWithDatasets} />
+      </Form>
+    );
+  }
 
   const defaultValues = {
     modelCategory: AiModelCategory.EM_NEURONS,
-    imageDataLayer: colorLayer.name,
   };
 
-  const segmentationLayers = getSegmentationLayers(dataset);
-  const fixedSelectedLayer = segmentationLayers.length === 1 ? segmentationLayers[0] : null;
-
-  const userBoundingBoxes = useSelector((state: OxalisState) =>
-    getUserBoundingBoxesFromState(state),
-  );
   const bboxesVoxelCount = _.sum(
-    userBoundingBoxes.map((bbox) => new BoundingBox(bbox.boundingBox).getVolume()),
+    (userBoundingBoxes || []).map((bbox) => new BoundingBox(bbox.boundingBox).getVolume()),
```

Comment:
While I'm reading this I thought of a potential issue. There is an assertion that there is at least one bounding box (later in the file). Now that multiple annotations can be supplied, it should be checked that each of them includes at least one bounding box - otherwise it does not make sense to supply that annotation since voxelytics won't use it for training.

Reply:
> While I'm reading this I thought of a potential issue. There is an assertion that there is at least one bounding box (later in the file). Now that multiple annotations can be supplied, it should be checked that each of them includes at least one bounding box - otherwise it does not make sense to supply that annotation since voxelytics won't use it for training.

That is a very good catch. In the current version the ai training modal is completely unaware of whether bounding boxes exist. 
Therefore, I added  the boxes of an annotation to `AnnotationWithDataset` which is a parameter of the `TrainAiModelTab`. It is now called `AnnotationWithDatasetAndBBoxes`. The tracings matching the annotations are now loaded on depend as soon as the csv input is validated to pass the bounding boxes to the `TrainAiModelTab` component.

Reply:
This should be done now :)